File: 2243-calculate-digit-sum-of-a-string/2243-calculate-digit-sum-of-a-string.py
class Solution(object):
    def digitSum(self, s, k):

        while len(s)>k:
            j=0
            n=[]
            for i in range(0,len(s),k):
                List1=s[i:i+k]
                Sum = sum(int(ch) for ch in List1)
                n.append(str(Sum))
            s="".join(n)
        return s


        
        

File: 2243-calculate-digit-sum-of-a-string/2243-calculate-digit-sum-of-a-string.java
class Solution {
    public String digitSum(String s, int k) {
        String S="";
        String S1="";
        while(s.length()>k){
            S1="";
            for(int i=0;i<s.length();i=i+k){ 
                if(s.length() < i+k){ 
                    int n =(s.length()%k);
                       S=s.substring(i,i+n);
                }else{                
                       S=s.substring(i,i+k);
                }
                int sum=0;
                for(char ch:S.toCharArray()){
                    sum=sum+(ch-'0');
                }
                S1=S1+sum;
            }
            s=S1;
        }
        return s;
    }
}